src/SelectPanel/SelectPanel.stories.tsx

```diff
@@ -54,7 +54,12 @@ export const Default = () => {
         title="Select labels"
         subtitle="Use labels to organize issues and pull requests"
         renderAnchor={({children, 'aria-labelledby': ariaLabelledBy, ...anchorProps}) => (
-          <Button trailingAction={TriangleDownIcon} aria-labelledby={` ${ariaLabelledBy}`} {...anchorProps}>
+          <Button
+            trailingAction={TriangleDownIcon}
+            aria-labelledby={` ${ariaLabelledBy}`}
+            {...anchorProps}
+            aria-haspopup="listbox"
+          >
```

Comment:
```suggestion
          <Button
            trailingAction={TriangleDownIcon}
            aria-labelledby={` ${ariaLabelledBy}`}
            aria-haspopup="dialog"
            {...anchorProps}
          >
```

Reply:
Hey @joshblack,

The order which you suggested will not work

```
aria-haspopup="dialog"
{...anchorProps}
```
because first it will take the value of aria-haspopup to `dialog` then it will overwrite the value with `true` from [anchorProps](https://github.com/primer/react/blob/main/src/AnchoredOverlay/AnchoredOverlay.tsx#L171). So to fix that we need first pass `anchorProps` and then `dialog`.

```
{...anchorProps}
aria-haspopup="dialog"
```

I have updated the screenshots in the PR description with the current changes feel free to take a look.

Also I found something while testing the changes locally that the same issue still exists in [SelectPanel.features.stories.tsx](https://github.com/primer/react/blob/main/src/SelectPanel/SelectPanel.features.stories.tsx#L55).  Let me know should we fix that in this PR or open a complete new issue for that.

Thanks!